src/main/java/Storage.java

```diff
@@ -0,0 +1,101 @@
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class Storage {
+    private final String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public String getFilePath() {
+        return this.filePath;
+    }
+
+    public void writeToFile(TaskList taskList) {
+        String output = "";
+        for (int i = 0; i < taskList.numOfTasks(); i++) {
+            output += taskList.getTask(i).toString() + "\n";
+        }
+        try {
+            BufferedWriter bw = new BufferedWriter(new FileWriter(this.filePath));
+            bw.write(output);
+            bw.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public TaskList readFromFile() {
+        TaskList taskList = new TaskList();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(this.filePath));
+            String input = br.readLine();
+            String type = "";
+            int length = 0;
+            int lastIndex = 0;
+            String timeWithBracket = "";
+            int twbLength = 0;
+            int twbLastIndex = 0;
+            String year = "";
+            String yearLine = "";
+            int descLastIndex = 0;
+            Task task = null;
+            String description = "";
+            String date = "";
+            String time = "";
```

Comment:
Perhaps you can consider abstracting away the formatting/extraction of dates/times by creating a separate class for functions related to these? There are a lot of variables declared here and abstracting these details will make the code easier to understand.

Reply:
Will take note, thanks.